bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/LedgerMetadataIndex.java

```diff
@@ -300,6 +300,9 @@ public void flush() throws IOException {
         int updatedLedgers = 0;
         while (!pendingLedgersUpdates.isEmpty()) {
```

Comment:
If this method is not thread-safe, perhaps we could use the `poll` method and determine if it is `null` to implement loop control.

Reply:
Make sense.